packages/react-router/lib/dom/global.ts

```diff
@@ -1,22 +1,12 @@
 import type { HydrationState, Router as DataRouter } from "../router/router";
-import type { AssetsManifest, CriticalCss, FutureConfig } from "./ssr/entry";
+import type { ServerHandoff } from "../server-runtime/serverHandoff";
+import type { AssetsManifest } from "./ssr/entry";
 import type { RouteModules } from "./ssr/routeModules";
 
-export type WindowReactRouterContext = {
-  basename?: string;
-  state: HydrationState;
-  criticalCss?: CriticalCss;
-  future: FutureConfig;
-  ssr: boolean;
-  isSpaMode: boolean;
+export type WindowReactRouterContext = ServerHandoff & {
```

Comment:
Now that we collapsed `@remix-run/react` and `@remix-run/server-runtime` we can share some of these types better.  The window context is exactly the server handoff plus the state/stream stuff we append to it after the fact